clientconn.go

```diff
@@ -196,18 +208,6 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 		defer cancel()
```

Comment:
Does this work if there's a timeout set (`cc.dopts.timeout > 0`) ?  In that case, the `defer`red actions will happen in the reverse order with this PR, so the auto-`cancel()` of the replaced `ctx` (with a timeout) happens before the check whether `ctx` is `Done()`, which means a cancellation error will always be returned.

Reply:
Right!
I filed #2736 to track. Will have a solution shortly!
Thanks for noting and mentioning this!!

Reply:
I've got a fix & testcase on its way.